src/tests/SimTest.cpp

```diff
@@ -152,33 +198,56 @@ void SimTest::reconfigure() {
   cfg2.activeSceneName = skokloster;
   simulator.reconfigure(cfg2);
   CORRADE_VERIFY(pathfinder != simulator.getPathFinder());
+
+  // test using MM ctor
+  SimulatorConfiguration cfg_mm;
+  cfg_mm.activeSceneName = vangogh;
+  MetadataMediator::ptr MM = MetadataMediator::create(cfg_mm);
+  Simulator simulator_mm(MM);
+  PathFinder::ptr pathfinder_mm = simulator_mm.getPathFinder();
+  simulator_mm.reconfigure(cfg_mm);
+  CORRADE_VERIFY(pathfinder_mm == simulator_mm.getPathFinder());
+  SimulatorConfiguration cfg2_mm;
+  cfg2_mm.activeSceneName = skokloster;
+  simulator_mm.reconfigure(cfg2_mm);
+  CORRADE_VERIFY(pathfinder_mm != simulator_mm.getPathFinder());
 }
 
 void SimTest::reset() {
+  auto testReset = [&](Simulator& simulator) {
+    PathFinder::ptr pathfinder = simulator.getPathFinder();
+    auto pinholeCameraSpec = SensorSpec::create();
+    pinholeCameraSpec->sensorSubType = esp::sensor::SensorSubType::Pinhole;
+    pinholeCameraSpec->sensorType = SensorType::Color;
+    pinholeCameraSpec->position = {0.0f, 1.5f, 5.0f};
+    pinholeCameraSpec->resolution = {100, 100};
+    AgentConfiguration agentConfig{};
+    agentConfig.sensorSpecifications = {pinholeCameraSpec};
+    auto agent = simulator.addAgent(agentConfig);
+
+    auto stateOrig = AgentState::create();
+    agent->getState(stateOrig);
+
+    simulator.reset();
+
+    auto stateFinal = AgentState::create();
+    agent->getState(stateFinal);
+    CORRADE_VERIFY(stateOrig->position == stateFinal->position);
```

Comment:
```suggestion
    CORRADE_COMPARE(stateOrig->position, stateFinal->position);
```

assuming those are Magnum vectors and not Eigen vectors (which don't have a debug output), same below

Reply:
Oh yeah. Ty

Reply:
oops, those are eigen vecs

Reply:
`CORRADE_COMPARE(Mn::Vector3{stateOrig->position}, Mn::Vector3{stateFinal->position});` ? :sweat_smile: